virtual_rainforest/models/animals/animal_communities.py

```diff
@@ -187,11 +206,14 @@ def collect_prey(self, consumer_cohort: AnimalCohort) -> list[AnimalCohort]:
             min_size, max_size = consumer_cohort.prey_groups[prey_functional_group]
 
             # Filter the potential prey cohorts based on their size
-            right_sized_prey = (
-                c for c in potential_prey_cohorts if min_size <= c.mass <= max_size
-            )
-            prey.extend(right_sized_prey)
-
+            for cohort in potential_prey_cohorts:
+                if min_size <= cohort.mass <= max_size:
+                    # Check if the cohort has zero individuals
+                    # if cohort.individuals <= 0:
+                    # Call the die_cohort method here
+                    #    self.die_cohort(cohort)
+                    #    continue  # Skip this cohort and move on to the next one
+                    prey.append(cohort)
```

Comment:
This is totally up to you, but you *could* just return the prey items lazily rather than copying them into a `list`. It would be faster in the case where the user just wanted the first prey item, for example.

To do this, you'd drop the `prey` list altogether and use this line instead:

```suggestion
                    yield cohort
```

The return type would then be `Iterable[AnimalCohort]`.

As I said, it's fine as is, but this would be an alternative way to do it.